lua/lspconfig/configs.lua

```diff
@@ -34,13 +34,11 @@ end
 ---@param config_name string
 ---@param config_def table Config definition read from `lspconfig.configs.<name>`.
 function configs.__newindex(t, config_name, config_def)
-  validate {
-    name = { config_name, 's' },
-    default_config = { config_def.default_config, 't' },
-    on_new_config = { config_def.on_new_config, 'f', true },
-    on_attach = { config_def.on_attach, 'f', true },
-    commands = { config_def.commands, 't', true },
-  }
+  validate('name', config_name, 'string')
+  validate('default_config', config_def.default_config, 'table')
+  validate('on_new_config', config_def.on_new_config, 'function', true)
+  validate('on_attach', config_def.on_attach, 'function', true)
+  validate('commands', config_def.commands, 'table', true)
```

Comment:
This form of validate() requires Nvim 0.11 https://github.com/neovim/neovim/commit/48251134ee59a3e2f0aeb89608fa820c21b25d4f